hmetrics/onload/init.go

```diff
@@ -43,29 +43,27 @@ const (
 )
 
 func init() {
+	var logger hmetrics.ErrHandler = func(_ error) error { return nil }
+
+	val := os.Getenv("HMETRICS_VERBOSE")
+	should, err := strconv.ParseBool(val)
+	if err == nil && should {
+		logger = func(err error) error {
+			log.Printf("[hmetrics] error: %v", err)
+			return nil
+		}
+	}
+
 	go func() {
 		var backoff int64
 		for backoff = 1; ; backoff++ {
 			start := time.Now()
 
-			var logger hmetrics.ErrHandler
-
-			val := os.Getenv("HMETRICS_VERBOSE")
-			should, err := strconv.ParseBool(val)
-			if err == nil && should {
-				logger = func(err error) error {
-					log.Printf("[hmetrics] error: %v", err)
-					return nil
-				}
-			}
-
 			err = hmetrics.Report(context.Background(), hmetrics.DefaultEndpoint, logger)
 			if time.Since(start) > 5*time.Minute {
 				backoff = 1
 			}
-			if logger != nil {
-				logger(err)
-			}
+			_ = logger(err)
```

Comment:
Why not `logger(err)` ?

Reply:
fixed in https://github.com/heroku/x/pull/25/commits/6ed42b46be7bbf1e35ee04b29d6f2e86aa04d04e